packages/js-drive/lib/abci/handlers/proposal/verifyChainLockFactory.js

```diff
@@ -0,0 +1,67 @@
+const ChainlockVerificationFailedError = require('../../errors/ChainlockVerificationFailedError');
+
+/**
+ *
+ * @param {RpcClient} coreRpcClient
+ * @param {BlockExecutionContext} latestBlockExecutionContext
+ * @param {BaseLogger} logger
+ * @return {verifyChainLock}
+ */
+function verifyChainLockFactory(
+  coreRpcClient,
+  latestBlockExecutionContext,
+  logger,
+) {
+  /**
+   * @typedef verifyChainLock
+   * @param {ChainLock} coreChainLock
+   * @return {Promise<void>}
+   */
+  async function verifyChainLock(coreChainLock) {
+    const lastCoreChainLockedHeight = latestBlockExecutionContext.getCoreChainLockedHeight();
+    if (coreChainLock.coreBlockHeight <= lastCoreChainLockedHeight) {
+      throw new ChainlockVerificationFailedError('coreBlockHeight is bigger than lastCoreChainLockedHeight', { chainlock: coreChainLock.toJSON(), lastCoreChainLockedHeight });
```

Comment:
It shouldn't be an internal error. You just need to invalidate block and write a message to logs with level debug